### PR TITLE
refactor: upgrade + optimise Redux

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Upgrades Redux to the latest version and optimises/improves type safety of middlewares.

## How this PR fixes it

- `react-redux` and `@reduxjs/toolkit` have been upgraded to the latest versions.
- All midleware (apart from persistence-related ones) have been migrated to [`createListenerMiddleware`](https://redux-toolkit.js.org/api/createListenerMiddleware)

## How to test it

No visible changes should be noticed:
- Added Safes in the sidebar should remain up-to-date (balance).
- Pending transaction signatures/executions should clear successfully.
- Pending messages should clear when signed successfully.

## Checklist
* [x] I've tested the branch on mobile 📱
* [x] I've documented how it affects the analytics (if at all) 📊
* [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/store/index.ts [hunk 1]
```diff
@@ -90,7 +88,10 @@ export const _hydrationReducer: typeof rootReducer = (state, action) => {
 const makeStore = (initialState?: Record<string, any>) => {
   return configureStore({
     reducer: _hydrationReducer,
-    middleware: (getDefaultMiddleware) => getDefaultMiddleware({ serializableCheck: false }).concat(middleware),
+    middleware: (getDefaultMiddleware) => {
+      listeners.forEach((listener) => listener(listenerMiddlewareInstance))
```
Comment (usame-algan) [inline src/store/index.ts:92]: Is there a difference in registering the listeners in this function or on the top level? In the examples from the documentation they are registered on the top level right after the middleware is created.

**Review by usame-algan — APPROVED** [review]
Looks good 👍 
I like the better type support and reduced nesting with the new middleware.

File: src/store/index.ts (hunk 1 above)
Comment (iamacook) [reply src/store/index.ts:92]: I don't think so.

I moved it here as this is just before store creation/allowed for better testing. What are your thoughts?

**Comment by francovenica** [comment]
LGTM

Checked that sidebar updates with the safes, new safes created, added safes, owned safes in new networks
Checked the lists queue history and message. Tx update just fine after creating/executing them
Tried with messages on and off chain, they update fine as well
Tried pagination, pagination with filter, tx updating after switching safes

**Comment by katspaugh** [comment]
@iamacook please don't merge just yet, let's create an RC w/o it first.

**Pull request merged**